osu.Game.Rulesets.Catch/Edit/CatchSelectionHandler.cs

```diff
@@ -52,16 +52,21 @@ public override bool HandleMovement(MoveSelectionEvent<HitObject> moveEvent)
             return true;
         }
 
-        public override bool HandleFlip(Direction direction)
+        public override bool HandleFlip(Direction direction, bool flipOverOrigin)
         {
+            if (SelectedItems.Count == 0 && !flipOverOrigin)
+                return false;
+
             var selectionRange = CatchHitObjectUtils.GetPositionRange(SelectedItems);
 
             bool changed = false;
+
             EditorBeatmap.PerformOnSelection(h =>
             {
                 if (h is CatchHitObject catchObject)
-                    changed |= handleFlip(selectionRange, catchObject);
+                    changed |= handleFlip(selectionRange, catchObject, flipOverOrigin);
```

Comment:
Thoughts on moving the computation of `selectionRange` into this `handleFlip()` call? Seems weird to pass a value that's never used in the scope of the method that's passing it. Again, not caused by this pull, but adding the `flipOverOrigin` param has amplified the weirdness of this for me.

Reply:
Wouldn't that cause it to be recalculated per object? Seems like it could get expensive if flipping a whole beatmap.

Reply:
Hm, ok, missed the for loop somehow. Fair point.